server/protocols/jmap/src/main/java/org/apache/james/jmap/model/JmapMDN.java

```diff
@@ -116,6 +131,55 @@ public MDNDisposition getDisposition() {
         return disposition;
     }
 
+    public Message generateMDNMessage(Message originalMessage, MailboxSession mailboxSession) throws ParseException, IOException, InvalidOriginMessageForMDNNotificationException {
+
+        User user = User.fromUsername(mailboxSession.getUser().getUserName());
+
+        return MDN.builder()
+            .report(generateReport(originalMessage, mailboxSession))
+            .humanReadableText(textBody)
+            .build()
+        .asMime4JMessageBuilder()
+            .setTo(getSenderAddress(originalMessage))
+            .setFrom(user.asString())
+            .setSubject(subject)
+            .setMessageId(MimeUtil.createUniqueMessageId(user.getDomainPart().orElse(null)))
+            .build();
+    }
+
+    public String getSenderAddress(Message originalMessage) throws InvalidOriginMessageForMDNNotificationException {
+        return OptionalUtils.or(
+            Optional.ofNullable(originalMessage.getSender()),
+            Optional.ofNullable(originalMessage.getFrom())
+                .map(MailboxList::stream).orElse(Stream.of())
+                .findAny())
+            .orElseThrow(() -> InvalidOriginMessageForMDNNotificationException.missingField("Sender"))
+            .getAddress();
+    }
+
+    @JsonIgnore
```

Comment:
Why do you need this annotation here and not on the _generateMDNMessage_ method?

Reply:
Probably not needed at all.